packages/@angular/cli/tasks/serve.ts

```diff
@@ -31,7 +31,15 @@ export default Task.extend({
     if (projectConfig.project && projectConfig.project.ejected) {
       throw new SilentError('An ejected project cannot use the build command anymore.');
     }
-    rimraf.sync(path.resolve(this.project.root, outputPath));
+
+    const contentBase = path.resolve(this.project.root, outputPath);
+    if (serveTaskOptions.useDist) {
+      if(!fs.existsSync(contentBase)) {
+        throw new SilentError('Dist directory not found');
```

Comment:
No we're not. This flag serves data from the existing dist folder if it exists. If it doesn't exist that's not possible. If the flag isn't used then it behaves the same as before (Deletes the dist folder and serves from memory)